### ISSUE TITLE
Duplicate FAR Reference and Declaration

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#23758]
Razor -> C# (LSP)
![image](https://user-images.githubusercontent.com/14852843/86855967-42ee3900-c070-11ea-9ba2-1478913fc010.png)
1 definition node + 2 references
 
C# -> Razor
![image](https://user-images.githubusercontent.com/14852843/86855996-4d103780-c070-11ea-9e6e-39a2c75cd624.png)
1 definition node + 1 reference

Notably, the Counter.DoSomeCounting() declaration is duplicated in the LSP FAR results window (as a definition node and a reference). This should only be shown once, as is the case with C# -> Razor.

**Comment by TanayParikh** [comment]
[Filed VSTS issue](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1153249).

**Comment by TanayParikh** [comment]
Assigned on their end.

**Comment by TanayParikh** [comment]
Backlogging this as it's pending on external action and doesn't require any direct action from us, for now. This is (relatively) lower priority as results are still shown mostly as expected (with 1 duplicated entry).

**Comment by TanayParikh** [comment]
Bringing this back in, LSP platform has resolved the issue. Will validate with the next build and close out this issue.

**Comment by TanayParikh** [comment]
Getting one definition + 1 reference node now:
![image](https://user-images.githubusercontent.com/14852843/89443891-ca79a700-d705-11ea-9f33-f51cf183ce6c.png)

Actual:
However, clicking on the definition node (razor -> C#) doesn't do anything. 

Expected:
Clicking on the definition node takes you to the definition.

**Comment by TanayParikh** [comment]
Validated the linking fix. Closing this out.
